commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/ShoppingListDraft.java

```diff
@@ -0,0 +1,40 @@
+package io.sphere.sdk.shoppinglists;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.sphere.sdk.annotations.FactoryMethod;
+import io.sphere.sdk.annotations.ResourceDraftValue;
+import io.sphere.sdk.customers.Customer;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.models.Reference;
+import io.sphere.sdk.types.CustomFieldsDraft;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+@JsonDeserialize(as = ShoppingListDraftDsl.class)
+@ResourceDraftValue(factoryMethods = {
+        @FactoryMethod(parameterNames = {"name"}),
+})
+public interface ShoppingListDraft {
+
+    LocalizedString getName();
+
+    @Nullable
+    LocalizedString getDescription();
+
+    @Nullable
+    String getKey();
+
+    @Nullable
+    Reference<Customer> getCustomer();
+
+    @Nullable
+    LocalizedString getSlug();
+
+    List<LineItemDraft> getLineItems();
```

Comment:
In the docu says it's optional, but I had a case too in the past where in an array it said optional when it was never the case. Have you checked that it's really never null, even when the shopping list is newly created? If in that case it's returning an empty array, then we have to report it to the docu so that they fix it.